src/QMCHamiltonians/CoulombPotential.h

```diff
@@ -117,35 +119,65 @@ struct CoulombPotential: public QMCHamiltonianBase
     else
 #endif
     {
-      const int* restrict M=d->M.data();
-      const int* restrict J=d->J.data();
-      for(int iat=0; iat<nCenters; ++iat)
+      if(d->DTType == DT_SOA)
+      {
+        for(size_t iat=1; iat<nCenters; ++iat)
+        {
+          const RealType* restrict dist=d->Distances[iat];
+          T q=Z[iat];
+          for(size_t j=0; j<iat; ++j)
+            res+=q*Z[j]/dist[j];
+        }
+      }
+      else
       {
-        T q=Z[iat];
-        for(int nn=M[iat]; nn<M[iat+1]; ++nn)
-          res+=q*Z[J[nn]]*d->rinv(nn);
+        const int* restrict M=d->M.data();
+        const int* restrict J=d->J.data();
+        for(int iat=0; iat<nCenters; ++iat)
+        {
+          T q=Z[iat];
+          for(int nn=M[iat]; nn<M[iat+1]; ++nn)
+            res+=q*Z[J[nn]]*d->rinv(nn);
+        }
       }
     }
     return res;
   }
 
 
-  inline T evaluateAB(const DistanceTableData* d, const ParticleScalar_t* restrict Za, const ParticleScalar_t* restrict Zb)
+  /** JNKIM: Need to check the precision */
```

Comment:
This is open boundary condition. I only checked the O atom calculation in a big box. There is no fundamental change comparing to the old code. JK probably concerns about the accuracy of reciprocal in MP.